diagram-viewer/src/main/java/com/powsybl/diagram/viewer/nad/NetworkAreaDiagramModel.java

```diff
@@ -97,18 +96,27 @@ public LayoutParameters getLayoutParameters() {
         return layoutParametersBean.getLayoutParameters();
     }
 
-    public LabelProvider getLabelProvider(Network network) {
-        return DEFAULT_LABEL_PROVIDER.equals(labelProvider.getValue()) ? new DefaultLabelProvider(network, getSvgParameters()) : null;
+    public LabelProviderFactory getLabelProviderFactory() {
+        return DEFAULT_LABEL_PROVIDER.equals(labelProvider.getValue())
+                ? DefaultLabelProvider::new
+                : null;
     }
 
-    public StyleProvider getStyleProvider(Network network) {
+    public StyleProviderFactory getStyleProviderFactory() {
         return TOPOLOGICAL_STYLE_PROVIDER.equals(styleProvider.getValue())
-                ? new TopologicalStyleProvider(network)
-                : new NominalVoltageStyleProvider(network);
+                ? TopologicalStyleProvider::new
+                : NominalVoltageStyleProvider::new;
     }
 
-    public LayoutFactory getLayoutFactory() {
-        return BASIC_LAYOUT.equals(layoutFactory.getValue()) ? new BasicForceLayoutFactory() : null;
+    public LayoutFactory getLayoutFactory(Network network) {
+        // FIXME : user must be able to customize positions
```

Comment:
indeed! myabe we could have an open file dialog then, to choose a data file. But we still have to define the format. As this feature is already needed now in the viewer, let's keep that for later